contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -260,10 +255,30 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, IERC677Rece
 
   // @inheritdoc IFunctionsSubscriptions
   function getSubscription(uint64 subscriptionId) public view override returns (Subscription memory) {
-    _isExistingSubscription(subscriptionId);
     return s_subscriptions[subscriptionId];
   }
 
+  // @inheritdoc IFunctionsSubscriptions
+  function getSubscriptionsInRange(
+    uint64 subscriptionIdStart,
+    uint64 subscriptionIdEnd
+  ) external view override returns (Subscription[] memory subscriptions) {
+    if (
+      subscriptionIdStart > subscriptionIdEnd ||
+      subscriptionIdEnd > s_currentSubscriptionId ||
+      s_currentSubscriptionId == 0
+    ) {
+      revert InvalidCalldata();
+    }
+
+    subscriptions = new Subscription[]((subscriptionIdEnd - subscriptionIdStart) + 1);
+    for (uint256 i = 0; i <= subscriptionIdEnd - subscriptionIdStart; ++i) {
```

Comment:
Can "i" be of type uint64? Then you can avoid a cast below and overflow can't happen anyway.

Reply:
Rens mentioned that in Solidity any loop iterator variable is always compiled as a `uint`.
I just did a quick search to find docs and that and couldn't find anything, so sounds like one of those hidden quirks.
I could dig deeper if we need the size. We are currently okay.